### ISSUE TITLE
ArgumentException: An item with the same key has already been added. Key: Strict-Transport-Security

### ISSUE THREAD

**Issue by hades200082** [juunas11/aspnetcore-security-headers#23]
When using `app.UseStatusCodePagesWithReExecute("/error/{0}")` I get this error whenever I `return NotFound()` from any controller.

in `/src/Joonasw.AspNetCore.SecurityHeaders/Hsts/HstsMiddleware.cs` on line 25 you need to check if the header has already been added...

```cs
if (context.Response.Headers.All(x => x.Key != HeaderName)) context.Response.Headers.Add(HeaderName, _headerValue);
```
You have the same issue in other places too - everywhere you are adding headers currently.

**Comment by hades200082** [comment]
For anyone else that hits this issue there is a workaround... `app.UseWhen()` will allow you to check the headers yourself and decide whether to run the middleware or not.

I think this should still be handled in the Hsts and Csp middleware automatically since the middleware can check if the values are correct more easily too and replace the header instead of trying to add a duplicate in that case.

```cs
app.UseWhen(context => context.Response.Headers.All(x => x.Key != "Strict-Transport-Security"),
    appBuilder =>
    {
        appBuilder.UseHsts(new HstsOptions(TimeSpan.FromDays(30), false, false));
    });

app.UseWhen(context => context.Response.Headers.All(x => x.Key != "Content-Security-Policy" && x.Key != "Content-Security-Policy-Report-Only"),
    appBuilder =>
    {
        appBuilder.UseCsp(csp =>
        {
            csp.ByDefaultAllow.FromSelf();

            // snip
        });
    });
```

**Comment by juunas11** [comment]
Will fix this, thanks for reporting :)

**Comment by juunas11** [comment]
These are now fixed for HSTS and HPKP, 2.6.0 will be available on NuGet soon.
